components/prism-wgsl.js

```diff
@@ -0,0 +1,94 @@
+Prism.languages.wgsl = {
+    'comment': {
+        pattern: /\/\/.*|\/\*[\s\S]*?(?:\*\/|$)/,
+        greedy: true,
+    },
+    'attributes': {
+        pattern: /(@)(?:(?:[_\p{XID_Start}][\p{XID_Continue}]+)|(?:[\p{XID_Start}]))/u,
+        lookbehind: true,
+        alias: 'symbol',
+    },
+    'functions': {
+        pattern: /\b(fn)\s+(?:(?:[_\p{XID_Start}][\p{XID_Continue}]+)|(?:[\p{XID_Start}]))(?=(?:\()|(?:<))/u,
+        lookbehind: true,
+        alias: 'function',
+        inside: {},
+    },
+    'keyword': [
+        { pattern: /\b(?:bitcast|block|break|case|continue|continuing|default|discard|else|elseif|enable|fallthrough|for|function|if|loop|private|read|read_write|return|storage|switch|uniform|workgroup|write)\b/ },
+        { pattern: /\b(?:asm|const|do|enum|handle|mat|premerge|regardless|typedef|unless|using|vec|void|while)\b/ },
+        { pattern: /\b(?:let|var)\b/ },
+        { pattern: /\b(?:type)\b/ },
+        { pattern: /\b(?:enum)\b/ },
+        { pattern: /\bfn\b/ },
+    ],
```

Comment:
Please combine all of those variants into a single pattern. The current categorization is somewhat arbitrary. Splitting this into multiple regexes also hinders our regex tooling.